code/datums/quirks/negative.dm

```diff
@@ -982,3 +982,134 @@
+
+/// Signal handler for when the badback quirk_holder equips an item
+/datum/quirk/old_man_jenkins/proc/on_equipped_item(mob/living/source, obj/item/equipped_item, slot)
+	SIGNAL_HANDLER
+
+	var/obj/item/left_hand_item = quirk_holder.held_items[1]
+	var/obj/item/right_hand_item = quirk_holder.held_items[2]
+	var/is_cane_left_hand = istype(left_hand_item) && HAS_TRAIT(left_hand_item, TRAIT_CANE_TOOL)
+	var/is_cane_right_hand = istype(right_hand_item) && HAS_TRAIT(right_hand_item, TRAIT_CANE_TOOL)
+	var/is_holding_cane = is_cane_left_hand || is_cane_right_hand
+
+	if(!is_holding_cane)
+		quirk_holder.add_movespeed_modifier(/datum/movespeed_modifier/bad_back_slowdown)
+		quirk_holder.AddElement(/datum/element/waddling)	
+	else if(is_holding_cane)
+		quirk_holder.remove_movespeed_modifier(/datum/movespeed_modifier/bad_back_slowdown)
+		quirk_holder.RemoveElement(/datum/element/waddling)
+
+	var/obj/item/primary_cane = left_cane?.resolve()
+	var/obj/item/secondary_cane = right_cane?.resolve()
+
+	if(is_cane_left_hand)
+		if(primary_cane && (primary_cane != left_hand_item))
+			UnregisterSignal(primary_cane, list(COMSIG_ITEM_DROPPED, COMSIG_PARENT_QDELETING, COMSIG_MOVABLE_MOVED))
+			left_cane = null
+		else if(!primary_cane)
+			left_cane = WEAKREF(left_hand_item)
+			primary_cane = left_hand_item
+			RegisterSignal(primary_cane, COMSIG_ITEM_DROPPED, .proc/on_unequipped_cane)
+			RegisterSignal(primary_cane, COMSIG_PARENT_QDELETING, .proc/on_unequipped_cane)
+			RegisterSignal(primary_cane, COMSIG_MOVABLE_MOVED, .proc/on_unequipped_cane)
+	else if(primary_cane) // no cane in hand, time to remove the signal and weakref
+		UnregisterSignal(primary_cane, list(COMSIG_ITEM_DROPPED, COMSIG_PARENT_QDELETING, COMSIG_MOVABLE_MOVED))
+		left_cane = null
+
+	if(is_cane_right_hand)
+		if(secondary_cane && (secondary_cane != right_hand_item))
+			UnregisterSignal(secondary_cane, list(COMSIG_ITEM_DROPPED, COMSIG_PARENT_QDELETING, COMSIG_MOVABLE_MOVED))
+			right_cane = null
+		else if(!secondary_cane)
+			right_cane = WEAKREF(right_hand_item)
+			secondary_cane = right_hand_item
+			RegisterSignal(secondary_cane, COMSIG_ITEM_DROPPED, .proc/on_unequipped_cane)
+			RegisterSignal(secondary_cane, COMSIG_PARENT_QDELETING, .proc/on_unequipped_cane)
+			RegisterSignal(secondary_cane, COMSIG_MOVABLE_MOVED, .proc/on_unequipped_cane)
+	else if(secondary_cane) // no cane in hand, time to remove the signal and weakref
+		UnregisterSignal(secondary_cane, list(COMSIG_ITEM_DROPPED, COMSIG_PARENT_QDELETING, COMSIG_MOVABLE_MOVED))
+		right_cane = null
+
+/// Signal handler for when the quirk_holder unequips an equipped cane
+/datum/quirk/old_man_jenkins/proc/on_unequipped_cane(obj/item/source)
+	SIGNAL_HANDLER
+
+	// if they have dual canes then no need to change mood/speed
+	var/both_hands_have_canes = left_cane?.resolve() && right_cane?.resolve()
+
+	if(!both_hands_have_canes)
+		quirk_holder.add_movespeed_modifier(/datum/movespeed_modifier/bad_back_slowdown)
```

Comment:
It's sus to me how many times this is repeating, a lot of this code repeats. Any places you can be making shared procs?

Reply:
Simplified it with https://github.com/tgstation/tgstation/pull/71470/commits/f3f3eadc10f674e64f41057130ed7eefa74ec715